File: 887. 鸡蛋掉落.py
class Solution(object):
    def superEggDrop(self, K, N):
        """
        :type K: int
        :type N: int
        :rtype: int
        """
        dp=[[1]*K,[0]*K]
        res=1
        j=1
        while max(dp[0][-1],dp[-1][-1])<N:
            dp[j][0]=dp[1-j][0]+1
            for i in range(1,K):
                dp[j][i]=dp[1-j][i]+dp[1-j][i-1]+1
            res+=1
            j=(j+1)%2
        return res

File: 887. 鸡蛋掉落.java
class Solution {
    public int superEggDrop(int K, int N) {
        if(K==0){
            return 0;
        }
        int[] dp=new int[K+1];
        int[] nextDp;
        for(int F=1; ; F++){
            nextDp=new int[K+1];
            for(int i=1;i<=K;i++){
                nextDp[i]=dp[i]+dp[i-1]+1;
                if(nextDp[i]>=N){
                    return F;
                }
            }
            dp=nextDp;
        }
        
    }
}